exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -275,83 +275,86 @@ private void consumeEntireNextValue() throws IOException {
   private void writeData(MapWriter map, FieldSelection selection, boolean moveForward) throws IOException {
     //
     map.start();
-    outside: while (true) {
-
-      JsonToken t;
-      if(moveForward){
-        t = parser.nextToken();
-      }else{
-        t = parser.getCurrentToken();
-        moveForward = true;
-      }
+    try {
```

Comment:
Yes, I am pretty sure that is the case. I don't remember why exactly I needed to put this in a try-finally block.